app/models/post.rb

```diff
@@ -1,5 +1,17 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
   has_many :likes
   has_many :comments
+
+  after_create :increment_user_post_count
+
+  # Increment user posts count
+  def increment_user_post_count
+    user.increment!(:posts_count)
```

Comment:
- [ ] Since the post belongs to an author, not a user we need to fix this as well.

- [ ] Also what you have in the schema is `posts_counter`

  ```
  author.increment!(:posts_counter)
  ```